cinema/views.py

```diff
@@ -56,3 +83,47 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        date_str = self.request.query_params.get("date")
+        if date_str:
+            _date = datetime.strptime(date_str, "%Y-%m-%d")
+            queryset = queryset.filter(show_time__date=_date)
+
+        movie = self.request.query_params.get("movie")
+        if movie:
+            queryset = queryset.filter(movie=movie)
+
+        if self.action == "list":
+            queryset = (queryset.select_related(
+                "cinema_hall", "movie"
+            ).prefetch_related("tickets")
+            )
```

Comment:
Format your code correctly, there is no need to put this expression inside `()` from the beginning